pallets/snarcos/src/systems.rs

```diff
@@ -0,0 +1,82 @@
+use ark_poly::polynomial::univariate::DensePolynomial;
+use ark_poly_commit::marlin_pc::MarlinKZG10;
+use ark_serialize::CanonicalDeserialize;
+use ark_snark::SNARK;
+use ark_std::rand::{prelude::StdRng, SeedableRng};
+use blake2::Blake2s;
+use codec::{Decode, Encode};
+use scale_info::TypeInfo;
+
+#[derive(Copy, Clone, Eq, PartialEq, Debug, Decode, Encode, TypeInfo)]
+pub enum ProvingSystem {
+    Groth16,
+    Gm17,
+    Marlin,
+}
+
+pub(super) trait VerifyingSystem {
+    type CircuitField: CanonicalDeserialize;
+    type Proof: CanonicalDeserialize;
+    type VerifyingKey: CanonicalDeserialize;
+
+    fn verify(
+        key: &Self::VerifyingKey,
+        input: &[Self::CircuitField],
+        proof: &Self::Proof,
+    ) -> Result<bool, ()>;
```

Comment:
I assume empty error here is a placeholder and something should be added in future. Maybe add a TODO here (with a reference to JIRA issue, if already created).